modules/apps/transfer/keeper/relay.go

```diff
@@ -69,6 +69,16 @@ func (k Keeper) sendTransfer(
 		return 0, errorsmod.Wrapf(channeltypes.ErrChannelNotFound, "port ID (%s) channel ID (%s)", sourcePort, sourceChannel)
 	}
 
+	appVersion, found := k.ics4Wrapper.GetAppVersion(ctx, sourcePort, sourceChannel)
+	if !found {
+		return 0, errorsmod.Wrapf(ibcerrors.ErrInvalidRequest, "application version not found for source port: %s and source channel: %s", sourcePort, sourceChannel)
+	}
+
+	// ics20-1 only supports a single coin, so if that is the current version, we must only process a single coin.
+	if appVersion == types.V1 && len(coins) > 1 {
+		return 0, errorsmod.Wrapf(ibcerrors.ErrInvalidRequest, "cannot transfer multiple coins with ics20-1")
+	}
+
```

Comment:
this looks like a duplicate of the `Transfer` fn, I wonder should we pull this out in the form of something like

`validateV1Coins(ctx, portID, channelID, coins) error` or similar WDYT?

Reply:
ah missed that it was being removed, nvm think it's fine as is!